### PR TITLE
[feat] Pull Posts from Database and display/render on screen

### PR DESCRIPTION
**Pull request by nathanhuynhers**
### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

- Integrated Supabase Fetching: Implemented fetchPostData() to retrieve posts from the Supabase posts table.
- Rendered the posts on screen with custom styling

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

image

## How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

- run pnpm run dev and inspect website changes

CC: @adityapawar1

### CODE REVIEW

File: app/PostScreen.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
+'use client';
+
+import { useEffect, useState } from 'react';
+import { supabase } from '@/supabase/client';
+import '../assets/global.css';
```
Comment (adityapawar1) [inline app/PostScreen.tsx:5]: I think you can remove this import, it might introduce some weird errors with styling since its importing all the global styles here, but they aren't being used. If a class name from the global css file conflicts with the local style.module.css, then the styling can be weird and its hard to debug.

File: app/PostItem.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
+import { FiHeart as HeartIcon } from 'react-icons/fi';
+import { GoPaperAirplane as ShareIcon } from 'react-icons/go';
+import { LuCircle as ProfileIcon } from 'react-icons/lu';
+import styles from './styles.module.css';
+import '../assets/global.css';
```
Comment (adityapawar1) [inline app/PostItem.tsx:5]: I think you can remove this import, it might introduce some weird errors with styling since its importing all the global styles here, but they aren't being used. If a class name from the global css file conflicts with the local style.module.css, then the styling can be weird and its hard to debug.

File: app/styles.module.css [hunk 3]
```diff
@@ -10,7 +10,89 @@
...
+  margin-right: 200px;
+  font-weight: bold;
+}
+
+.date {
+  font-weight: 200;
+}
+
+.postImage {
+  width: 320px;
```
Comment (adityapawar1) [inline app/styles.module.css:71]: This should be a `width: 100%`, so the image spans the full width of the container.

**Review by adityapawar1 — CHANGES_REQUESTED** [review]
LOOKS AMAZING!!!! Couple of small changes left!
